### PR TITLE
mtdata downloader

### LINKED ISSUE #6: Corpus finder and downloader
We should include a GUI that helps download corpora automatically. We can mostly take the downloaders from here: https://github.com/mozilla/firefox-translations-training/tree/main/pipeline/data/importers/corpus but they need a few fixes (the opus and flores ones don't deal well with secondary language tags such as `zh_tw`.

The same is valid for the corpora finder script: https://github.com/mozilla/firefox-translations-training/blob/main/utils/find-corpus.py which I can start improving.

Not exactly sure how this would look in the GUI. Maybe a popup/ or a separate tag?

### PR DESCRIPTION
**Pull request by XapaJIaMnu**
This is just a draft as I am not exactly sure how to hook it up to the GUI, but I have coded the backend bits necessary for 

1) Deduplicating datasets (only taking the latest version of a dataset)
2) Downloading datasets (in parallel) using the mtdata cli.

I imagine it should be some "tab" such as "discover" datasets, where we get a list of them and we can manually exclude some/label them clean/medium/dirty. The downloader automatically splits train/test/dev based on the dataset id provided by mtdata.

This should fix #6 eventually.

### CODE REVIEW

File: mtdata-stuff.py [hunk 1]
```diff
@@ -53,3 +94,10 @@ def read_dataset(did: str):
 @app.get("/datasets/{did}/sample")
```
Comment (XapaJIaMnu) [inline mtdata-stuff.py:94]: This is duplicated with the function above. I assume that this is part of the javascript interface. @jelmervdl is that how it's supposed to work?

**Comment by XapaJIaMnu** [comment]
I propose we merge this for now so that it's in the tree and it can be picked up when we have a downloader interface.

**Comment by jelmervdl** [comment]
Stand-alone this doesn't add much actual functionality. I'll take this pull request over, and add some sort of minimal interface for it at least before merging it.

**Comment by jelmervdl** [comment]
Current route:
image

I'm really considering adding Bootstrap to the project for a bit more UI. Also a filter toggle for only the latest version of each dataset.

**Comment by XapaJIaMnu** [comment]
Looks good to me, pending on the answer of https://github.com/thammegowda/mtdata/issues/129 . Ideally i'd like to see how much i am downloading before starting to download (and also we had the issue with the mozilla pipeline that downloads would fail because we would be throttled, so we should be able to limit the number of parallel downloads.)

Cheers,

Nick

**Comment by jelmervdl** [comment]
Downloads are currently limited to two concurrent downloads.

Maybe I can get it do a HEAD request to get the size (or Content-Length really) of the download. It would be infeasible to do this for all datasets that are listed, but should be doable for the ones in your "shopping list" at least.

**Pull request merged**